test/ecto/model/dependent_test.exs

```diff
@@ -0,0 +1,121 @@
+Code.require_file "../../support/mock_repo.exs", __DIR__
```

Comment:
We already have all the associations in there anyway, we just need to set different dependent options.